File: test_vuln.go
package main

import (
	"database/sql"
	"fmt"
	"os/exec"
)

func login(db *sql.DB, username, password string) {
	// SQL Injection
	query := fmt.Sprintf("SELECT * FROM users WHERE username = '%s' AND password = '%s'", username, password)
	db.Query(query)
}

func runCommand(cmd string) {
	// Command Injection
	exec.Command("sh", "-c", cmd).Run()
}

File: test_vuln.java
import java.sql.*;

public class UserLogin {
    public void login(String username, String password) throws SQLException {
        Connection conn = DriverManager.getConnection("jdbc:mysql://localhost/db", "root", "pass");
        // SQL Injection
        String query = "SELECT * FROM users WHERE username = '" + username + "' AND password = '" + password + "'";
        Statement stmt = conn.createStatement();
        ResultSet rs = stmt.executeQuery(query);
    }

    public void runCommand(String cmd) throws Exception {
        // Command Injection
        Runtime.getRuntime().exec(cmd);
    }
}
